### PR TITLE
Add opendata.bordeaux-metropole.fr source

### LINKED ISSUE #3062: [Source Request]: Bordeaux Metropole, France
### Municipality / Region

Bordeaux Metropole, France

### Collection Calendar Webpage

https://opendata.bordeaux-metropole.fr/explore/dataset/en_frcol_s/export/?disjunctive.jour_col

### Example Address

Latitude, Longitude : 44.840165649597424, -0.6395375878056988

### Collection Data Format

As documented API, Something different (add to additional information)

### Additional Information

Webservice : https://opendata.bordeaux-metropole.fr/explore/dataset/en_frcol_s/webservices/?disjunctive.jour_col
Needed to find the coordinates inside a shape.

### PR DESCRIPTION
**Pull request by AthAshino**
Closes #3062

### CODE REVIEW

File: custom_components/waste_collection_schedule/waste_collection_schedule/source/opendata_bordeauxmetropole_fr.py [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+                    next_date = self._get_next_weekday(source_date, _day)
+                    entries.append(
+                        Collection(
+                            date=next_date,  # Next collection date
+                            t=LABEL_MAP.get(_collection_type, _collection_type),  # Collection type
+                            icon=ICON_MAP.get(_collection_type),  # Collection icon
+                        )
+                    )
+                    source_date = next_date
+                    if next_date == source_date:
```
Comment (5ila5) [inline custom_components/waste_collection_schedule/waste_collection_schedule/source/opendata_bordeauxmetropole_fr.py:335]: if is always True

Comment (AthAshino) [reply custom_components/waste_collection_schedule/waste_collection_schedule/source/opendata_bordeauxmetropole_fr.py]: Oups, fixed

**Comment by 5ila5** [comment]
Thanks for your contribution.

I did some reformatting (.pre-commit-config.yaml) and fixed the type hint issues reported by mypy

**Comment by AthAshino** [comment]
Great thanks! Merry Christmas

**Pull request merged**